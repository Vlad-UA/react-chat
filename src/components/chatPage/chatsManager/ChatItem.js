import React from 'react';

import {ListItem, ListItemText} from 'material-ui/List';

import AvatarItem from "../common/Avatar";

const ChatItem = ({title}) => {
  return (
    <ListItem button>
      <AvatarItem title={title} lettersQuantity={2} colorFrom={title}/>
      <ListItemText primary={title}/>
    </ListItem>
  );
}

export default ChatItem;
