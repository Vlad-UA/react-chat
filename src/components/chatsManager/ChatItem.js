import React from 'react';

import {ListItem, ListItemText} from 'material-ui/List';

import AvatarItem from "../AvatarItem";

const ChatItem = props => {
  const {title} = props;

  return (
    <ListItem button>
      <AvatarItem title={title} lettersQuantity={2}/>
      <ListItemText primary={title}/>
    </ListItem>
  );
}

export default ChatItem;
