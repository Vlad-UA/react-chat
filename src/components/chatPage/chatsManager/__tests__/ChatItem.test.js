/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ChatItem from '../ChatItem';

jest.mock('../../common/Avatar', () => () => 'Avatar');

const mockProps = {
  title: 'title',
  chatId: 'chatId',
  isActive: false,
  createdAt: '2018-05-25T12:20:19.920Z',
  disabled: false,
};

describe('<ChatItem />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <ChatItem {...mockProps} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('snapshot renders correctly - renders correctly', () => {
    const tree =
      renderer.create(<MemoryRouter><ChatItem {...mockProps} /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot renders correctly - renders disabled', () => {
    const tree =
      renderer
        .create(<MemoryRouter><ChatItem {...mockProps} disabled /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot renders correctly - renders active', () => {
    const tree =
      renderer
        .create(<MemoryRouter><ChatItem {...mockProps} active /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
