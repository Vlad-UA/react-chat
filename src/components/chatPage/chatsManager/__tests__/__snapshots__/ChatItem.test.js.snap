// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatItem /> snapshot renders correctly - renders active 1`] = `
<a
  className="MuiButtonBase-root-63 MuiListItem-root-53 MuiListItem-default-56 MuiListItem-gutters-60 MuiListItem-button-61"
  href="/chat/chatId"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  Avatar
  <div
    className="MuiListItemText-root-66"
  >
    <h3
      className="MuiTypography-root-72 MuiTypography-subheading-79 MuiListItemText-primary-69"
    >
      title
    </h3>
    <p
      className="MuiTypography-root-72 MuiTypography-body1-81 MuiTypography-colorTextSecondary-94 MuiListItemText-secondary-70"
    >
      19 days ago
    </p>
  </div>
  <span
    className="MuiTouchRipple-root-96"
  />
</a>
`;

exports[`<ChatItem /> snapshot renders correctly - renders correctly 1`] = `
<a
  className="MuiButtonBase-root-63 MuiListItem-root-53 MuiListItem-default-56 MuiListItem-gutters-60 MuiListItem-button-61"
  href="/chat/chatId"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  Avatar
  <div
    className="MuiListItemText-root-66"
  >
    <h3
      className="MuiTypography-root-72 MuiTypography-subheading-79 MuiListItemText-primary-69"
    >
      title
    </h3>
    <p
      className="MuiTypography-root-72 MuiTypography-body1-81 MuiTypography-colorTextSecondary-94 MuiListItemText-secondary-70"
    >
      19 days ago
    </p>
  </div>
  <span
    className="MuiTouchRipple-root-96"
  />
</a>
`;

exports[`<ChatItem /> snapshot renders correctly - renders disabled 1`] = `
<a
  className="MuiButtonBase-root-63 MuiButtonBase-disabled-64 MuiListItem-root-53 MuiListItem-default-56 MuiListItem-gutters-60 MuiListItem-disabled-58 MuiListItem-button-61"
  href="/chat/chatId"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="-1"
>
  Avatar
  <div
    className="MuiListItemText-root-66"
  >
    <h3
      className="MuiTypography-root-72 MuiTypography-subheading-79 MuiListItemText-primary-69"
    >
      title
    </h3>
    <p
      className="MuiTypography-root-72 MuiTypography-body1-81 MuiTypography-colorTextSecondary-94 MuiListItemText-secondary-70"
    >
      19 days ago
    </p>
  </div>
</a>
`;
